File: src/api/internal/handler/dto/datatemplate.dto.go
// Package dto for handling Data Transfer Objects for API requests and responses
package dto

import "excalibur/internal/handler/hateoas"

// DataTemplateColumnDTO represents a column of a DataTemplate
type DataTemplateColumnDTO struct {
	OriginalName string `json:"originalName"`
	PrettyName   string `json:"prettyName"`
	DataType     string `json:"dataType"`
}

// DataTemplateDTO represents the a data template structure
type DataTemplateDTO struct {
	Name    string                  `json:"name"`
	Columns []DataTemplateColumnDTO `json:"columns"`
}

// DataTemplateWithHateoasDTO for returning a result with resource pointer
type DataTemplateWithHateoasDTO struct {
	DataTemplateID string `json:"id"`
	hateoas.H
}

File: src/Excalibur.Api/DTOs/DataTemplateDto.cs
using Excalibur.Api.Models;

namespace Excalibur.Api.DTOs;

public class DataTemplateDto
{
    public string? Id { get; set; }
    public string Name { get; set; }
    public List<DataTemplateColumnDto> Columns { get; set; }

    public DataTemplate MapToModel()
    {
        return new DataTemplate
        {
            Id = Id,
            Name = Name,
            Columns = Columns.Select(c => new DataTemplateColumn 
            { 
                OriginalName = c.OriginalName, 
                PrettyName = c.PrettyName,
                DataType = c.DataType
            }).ToList(),
        };
    }
}
